src/main/java/net/openhft/chronicle/core/io/TracingReferenceCounted.java

```diff
@@ -85,40 +97,32 @@ private boolean tryReserve(ReferenceOwner id, boolean must) throws IllegalStateE
             }
             StackTrace stackTrace = references.get(id);
             if (stackTrace == null)
-                references.putIfAbsent(id, stackTrace("reserve", id));
+                references.put(id, stackTrace("reserve", id));
```

Comment:
Not sure why this needed to be putIfAbsent given we're in a `synchronized(references)` block and we establish that there's no entry on the previous line